inventory_management_system_api/services/item.py

```diff
@@ -118,9 +121,69 @@ def get(self, item_id: str) -> Optional[ItemOut]:
         :param item_id: The ID of the item to retrieve
         :return: The retrieved item, or `None` if not found
         """
-
         return self._item_repository.get(item_id)
 
+    def update(self, item_id: str, item: ItemPatchRequestSchema) -> ItemOut:
+        """
+        Update an item by its ID.
+
+        The method checks if the item exists in the database and raises a `MissingRecordError` if it does
+        not. If the catalogue item ID or system ID is being updated, it checks if a catalogue item
+        and/or system ID with such ID exists and raises a `MissingRecordError` if it does not.
+
+        :param item_id: The ID of the item to update.
+        :param item: The item containing the fields that need to be updated.
+        :return: The updated item.
+        """
+        update_data = item.model_dump(exclude_unset=True)
+
+        stored_item = self.get(item_id)
+        if not stored_item:
+            raise MissingRecordError(f"No item found with ID: {item_id}")
+
+        catalogue_item = None
+        if "catalogue_item_id" in update_data and item.catalogue_item_id != stored_item.catalogue_item_id:
+            catalogue_item = self._catalogue_item_repository.get(item.catalogue_item_id)
+            if not catalogue_item:
+                raise MissingRecordError(f"No catalogue item found with ID: {item.catalogue_item_id}")
+
+            if "properties" not in update_data:
+                item.properties = [PropertyPostRequestSchema(**prop.model_dump()) for prop in stored_item.properties]
+                update_data = item.model_dump(exclude_unset=True)
+
+        if "system_id" in update_data and item.system_id != stored_item.system_id:
+            system = self._system_repository.get(item.system_id)
+            if not system:
+                raise MissingRecordError(f"No system found with ID: {item.system_id}")
+
+        if "properties" in update_data:
+            if not catalogue_item:
+                catalogue_item = self._catalogue_item_repository.get(stored_item.catalogue_item_id)
```

Comment:
Would be nice to have comment(s) here (similar to the ones I left in the `catalogue_item` service module to explain what's going on.